PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx

```diff
@@ -304,700 +180,519 @@ void AliAnalysisTRDEfficiency::UserCreateOutputObjects()
-    fhgevent2   = new THnSparseD("fhgevent2", "fhgevent2",         dim, bins, xmin, xmax);  // name, title, dim, nbins, xmin, xmax
-    fhgevent3   = new THnSparseD("fhgevent3", "fhgevent3",         dim, bins, xmin, xmax);  // name, title, dim, nbins, xmin, xmax
-    fhgevent4   = new THnSparseD("fhgevent4", "fhgevent4",         dim, bins, xmin, xmax);  // name, title, dim, nbins, xmin, xmax
-    fhgevent5   = new THnSparseD("fhgevent5", "fhgevent5",         dim, bins, xmin, xmax);  // name, title, dim, nbins, xmin, xmax
-    fhgevent6   = new THnSparseD("fhgevent6", "fhgevent6",         dim, bins, xmin, xmax);  // name, title, dim, nbins, xmin, xmax
-    fhgevent7   = new THnSparseD("fhgevent7", "fhgevent7",         dim, bins, xmin, xmax);  // name, title, dim, nbins, xmin, xmax
-    fhgevent8   = new THnSparseD("fhgevent8", "fhgevent8",         dim, bins, xmin, xmax);  // name, title, dim, nbins, xmin, xmax
-    fhgevent9   = new THnSparseD("fhgevent9", "fhgevent9",         dim, bins, xmin, xmax);  // name, title, dim, nbins, xmin, xmax
-    
-    fhgevent2->GetAxis(0)->SetTitle("trd pT (GeV)");        // all 8 THnSparseD graphs have the same axis' names
-    fhgevent2->GetAxis(1)->SetTitle("trd pid");
-    fhgevent2->GetAxis(2)->SetTitle("gamma pt");
-    fhgevent2->GetAxis(3)->SetTitle("gamma R");
-    fhgevent2->GetAxis(4)->SetTitle("other trd pT (GeV)");
-    fhgevent2->GetAxis(5)->SetTitle("other trd pT (GeV)");
-
-    
-    fhgevent    = new TH1F("fhgevent", "fhgevent",  10, 0, 10);
-    fhevent     = new TH1F("fhevent", "fhevent",    10, 0, 10);
-    
-     dim = 4;          // dimensions
-//     Int_t bins1[4]     = {10000, 10000};   // # of bins
-//	 Int_t xmin1[4]     = {0, 0};           //         min
-//	 Int_t xmax1[4]     = {10000, 10000};  // max
-    //fhtrckvnt   = new THnSparseD("fhtrckvnt", "fhtrckvnt",          dim, bins1, xmin1, xmax1);
-    //fhtrckvnthqu= new THnSparseD("fhtrckvnthqu", "fhtrckvnthqu",    dim, bins1, xmin1, xmax1);
-    
-    fhtrvnt     = new TH1F("fhtrvnt", "fhtrvnt",        20000, 0, 20000);
-    fhtrvnthqu  = new TH1F("fhtrvnthqu", "fhtrvnthqu",  20000, 0, 20000);
-    
-    lsttrckvnt   = new TList();
-    lsttrckvnthqu= new TList();
-    
-    fhgetaphi1  = new TH2F("fhgetaphi1", "fhgetaphi1",  400, -1, 7, 400, -2, 2);
-    fhgR1       = new TH1F("fhgR1", "fhgR1",            2000, 0, 600);
-    fhgpt1      = new TH1F("fhgpt1", "fhgpt1",          2000, 0, 200);
-    fhgetaphi5  = new TH2F("fhgetaphi5", "fhgetaphi5",  800, -1, 7, 400, -2, 2);
-    fhgetaphi8  = new TH2F("fhgetaphi8", "fhgetaphi8",  800, -1, 7, 400, -2, 2);
-    fhgetaphi9  = new TH2F("fhgetaphi9", "fhgetaphi9",  800, -1, 7, 400, -2, 2);
-    
-    //fOutputList->Add(fHistPt);          // don't forget to add it to the list! the list will be written to file, so if you want
+    fHistPt = new TH1F("fHistPt", "fHistPt",5000, 0, 5000);       // create your histogram
+    
+    v0reader = new AliV0ReaderV1();

```

Comment:
is there a special reason not to use the global V0Reader, which we use for all our tasks? (sorry, I did not compare actual settings, maybe it is obvious then please let me know)

Reply:
And could you please fix the following compilation warnings:

/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx: In member function 'virtual void AliAnalysisTRDEfficiency::UserExec(Option_t*)':
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:394:26: warning: unused variable 'lstp' [-Wunused-variable]
                 Double_t lstp[4] = {trdtrack->Pt(), ptrack->Pt(), photon->GetConversionRadius(), rating};
                          ^~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:405:26: warning: unused variable 'lstn' [-Wunused-variable]
                 Double_t lstn[4] = {trdtrack->Pt(), ntrack->Pt(), photon->GetConversionRadius(), rating};
                          ^~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:328:23: warning: unused variable 'eventcuts' [-Wunused-variable]
     AliConvEventCuts *eventcuts = v0reader->GetEventCuts();
                       ^~~~~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:339:11: warning: unused variable 'year' [-Wunused-variable]
     Int_t year=2018;
           ^~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:353:12: warning: unused variable 'check' [-Wunused-variable]
     Bool_t check = convsel->ProcessEvent(lst, fESD, NULL);    
            ^~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:356:14: warning: unused variable 'cint7' [-Wunused-variable]
     Double_t cint7 = clss.Contains("CINT7-T");
              ^~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:357:14: warning: unused variable 'hqu' [-Wunused-variable]
     Double_t hqu   = clss.Contains("HQU");
              ^~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:358:14: warning: unused variable 'trd' [-Wunused-variable]
     Double_t trd   = 0;
              ^~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:359:14: warning: unused variable 'pt' [-Wunused-variable]
     Double_t pt, pid, sag, trcklt, lyr0 = 0;
              ^~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:359:18: warning: unused variable 'pid' [-Wunused-variable]
     Double_t pt, pid, sag, trcklt, lyr0 = 0;
                  ^~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:359:23: warning: unused variable 'sag' [-Wunused-variable]
     Double_t pt, pid, sag, trcklt, lyr0 = 0;
                       ^~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:359:28: warning: unused variable 'trcklt' [-Wunused-variable]
     Double_t pt, pid, sag, trcklt, lyr0 = 0;
                            ^~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:359:36: warning: unused variable 'lyr0' [-Wunused-variable]
     Double_t pt, pid, sag, trcklt, lyr0 = 0;
                                    ^~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx: In member function 'virtual Bool_t AliAnalysisTRDEfficiency::checkPi0(TClonesArray*, AliAODConversionMother*, Double_t*, Int_t)':
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:555:22: warning: unused variable 'lstp' [-Wunused-variable]
             Double_t lstp[4] = {trdtrack->Pt(), ptrack->Pt(), photon->GetConversionRadius(), rating};
                      ^~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:573:22: warning: unused variable 'lstn' [-Wunused-variable]
             Double_t lstn[4] = {trdtrack->Pt(), ntrack->Pt(), photon->GetConversionRadius(), rating};
                      ^~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:533:11: warning: unused variable 'index' [-Wunused-variable]
     Int_t index  = 0;
           ^~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:545:14: warning: unused variable 'ghqu' [-Wunused-variable]
     Double_t ghqu = 0;
              ^~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:546:14: warning: unused variable 'gcnt' [-Wunused-variable]
     Double_t gcnt = 0;
              ^~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx: In member function 'virtual Double_t AliAnalysisTRDEfficiency::GetRating(AliESDv0*, AliESDtrack*, AliESDTrdTrack*)':
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:684:23: warning: variable 'R' set but not used [-Wunused-but-set-variable]
     Double_t x, y, z, R;
                       ^
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx: In member function 'virtual Bool_t AliAnalysisTRDEfficiency::checkPi0(TClonesArray*, AliAODConversionMother*, Double_t*, Int_t)':
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:599:12: warning: 'ntrd' may be used uninitialized in this function [-Wmaybe-uninitialized]
     tmp[10]= ntrd;
     ~~~~~~~^~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:598:12: warning: 'nrat' may be used uninitialized in this function [-Wmaybe-uninitialized]
     tmp[9] = nrat;
     ~~~~~~~^~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:597:12: warning: 'nsag' may be used uninitialized in this function [-Wmaybe-uninitialized]
     tmp[8] = nsag;
     ~~~~~~~^~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:596:12: warning: 'npid' may be used uninitialized in this function [-Wmaybe-uninitialized]
     tmp[7] = npid;
     ~~~~~~~^~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:595:12: warning: 'npt' may be used uninitialized in this function [-Wmaybe-uninitialized]
     tmp[6] = npt;
     ~~~~~~~^~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:592:12: warning: 'ptrd' may be used uninitialized in this function [-Wmaybe-uninitialized]
     tmp[4] = ptrd;
     ~~~~~~~^~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:591:12: warning: 'prat' may be used uninitialized in this function [-Wmaybe-uninitialized]
     tmp[3] = prat;
     ~~~~~~~^~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:590:12: warning: 'psag' may be used uninitialized in this function [-Wmaybe-uninitialized]
     tmp[2] = psag;
     ~~~~~~~^~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:589:12: warning: 'ppid' may be used uninitialized in this function [-Wmaybe-uninitialized]
     tmp[1] = ppid;
     ~~~~~~~^~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:588:12: warning: 'ppt' may be used uninitialized in this function [-Wmaybe-uninitialized]
     tmp[0] = ppt;
     ~~~~~~~^~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx: In member function 'virtual void AliAnalysisTRDEfficiency::UserExec(Option_t*)':
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:431:65: warning: 'nrat' may be used uninitialized in this function [-Wmaybe-uninitialized]
                                gpt, gR  , geta, gphi, ghqu, gcnt};
                                                                 ^
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:431:65: warning: 'nsag' may be used uninitialized in this function [-Wmaybe-uninitialized]
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:431:65: warning: 'npid' may be used uninitialized in this function [-Wmaybe-uninitialized]
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:431:65: warning: 'npt' may be used uninitialized in this function [-Wmaybe-uninitialized]
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:431:65: warning: 'prat' may be used uninitialized in this function [-Wmaybe-uninitialized]
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:431:65: warning: 'psag' may be used uninitialized in this function [-Wmaybe-uninitialized]
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:431:65: warning: 'ppid' may be used uninitialized in this function [-Wmaybe-uninitialized]
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx:431:65: warning: 'ppt' may be used uninitialized in this function [-Wmaybe-uninitialized]

Reply:
The reason I use the V0Reader like this is because it doesn't crash the program. I am probably implementing it incorrectly.

Reply:
If I can help please let me know

Reply:
Thank you, but I have solved the issue; I had forgotten to load the V0Reader in the AddTask.